ignite/contrib/handlers/tensorboard_logger.py

```diff
@@ -0,0 +1,227 @@
+                                                        another_engine=trainer),
+                             event_name=Events.EPOCH_COMPLETED)
+
+            # Attach the logger to the evaluator on the validation dataset and log NLL, Accuracy metrics after
+            # each epoch. We setup `another_engine=trainer` to take the epoch of the `trainer` instead of `evaluator`.
+            tb_logger.attach(evaluator,
+                             log_handler=output_handler(tag="validation",
+                                                        metric_names=["nll", "accuracy"],
+                                                        another_engine=trainer),
+                             event_name=Events.EPOCH_COMPLETED)
+
+            # Attach the logger to the trainer to log optimizer's parameters, e.g. learning rate at each iteration
+            tb_logger.attach(trainer,
+                             log_handler=optimizer_params_handler(optimizer),
+                             event_name=Events.ITERATION_COMPLETED)
+
+            # Log model's graph
+            x, _ = next(iter(train_loader))
+            tb_logger.log_graph(model, x.to(device))
+
+            # Attach the logger to the trainer to log model's weights norm after each iteration
+            tb_logger.attach(trainer,
+                             log_handler=weights_scalar_handler(model),
+                             event_name=Events.ITERATION_COMPLETED)
+
+            # Attach the logger to the trainer to log model's weights as a histogram after each epoch
+            tb_logger.attach(trainer,
+                             log_handler=weights_hist_handler(model),
+                             event_name=Events.EPOCH_COMPLETED)
+
+            # Attach the logger to the trainer to log model's gradients norm after each iteration
+            tb_logger.attach(trainer,
+                             log_handler=grads_scalar_handler(model),
+                             event_name=Events.ITERATION_COMPLETED)
+
+            # Attach the logger to the trainer to log model's gradients as a histogram after each epoch
+            tb_logger.attach(trainer,
+                             log_handler=grads_hist_handler(model),
+                             event_name=Events.EPOCH_COMPLETED)
+
+    """
+
+    def __init__(self, log_dir):
+        self.writer = SummaryWriter(log_dir=log_dir)
+
+    def _close(self):
+        self.writer.close()
+
+    def __del__(self):
+        self._close()
+
+    def log_graph(self, model, input_to_model, **kwargs):
+        """Log model's graph
+
+        Args:
+            model (torch.nn.Module): input model
+            input_to_model (torch.Tensor or torch.utils.data.DataLoader): input to the model
+            **kwargs: kwargs to `tensorboardX.SummaryWriter.add_graph`
+
+        """
+        from torch.utils.data import DataLoader
+
+        if isinstance(input_to_model, DataLoader):
+            input_to_model, _ = next(iter(input_to_model))
```

Comment:
Here is my comment copied from the *old* tensorboard_logger (#265). I just realized that this new logger exists 👍:
Assuming that seeds are set manually at the beginning of an experiment, this line of code can break reproducibility. Running the same experiment with with and without logging the graph can result in different batch data passed to the engines `process_function` because the dataloader could be called once here. Perhaps something like this works:
```
x, y = input_to_model.collate_fn(input_to_model.dataset.__getitem__(0))
```

Reply:
@Bibonaut yes you're right about the batches. Concerning your proposition, probably should be something like:
```
x, _ = input_to_model.collate_fn([input_to_model.dataset[0], ])
```
Honestly, I think we should remove such option to take a dataloader as an input...

Reply:
What if we take input shape as a parameter instead of the dataloader? We can create a tensor of zeros to pass through the model

Reply:
This would work if we have a model working directly on tensors, but it could have more than one entry. I think we need to remove this method `log_graph` as it adds nothing more special to `tb_logger.writer.add_graph(model, inputs)`